### PR TITLE
Add html demo for async generators.

### PR DESCRIPTION
**Pull request by mnieper**
After having written a section for the wiki, I propose this PR as a simple demo for async generators and for-on loops.

### CODE REVIEW

File: test/feature/AsyncGenerators/Demo.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Options: --async-functions --async-generators --for-on
+// Async.
+
+function timeout(ms) {
```
Comment (arv) [inline test/feature/AsyncGenerators/Demo.js]: Can we use requestAnimationFrame for this demo?

Comment (arv) [inline test/feature/AsyncGenerators/Demo.js]: Never mind... I didn't realize this was a test.

File: test/feature/AsyncGenerators/Demo.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
+}
+
+async function* asyncStream() {
+  var i = 0;
+  while (true) {
+    await timeout(50);
```
Comment (arv) [inline test/feature/AsyncGenerators/Demo.js]: please reduce time to make test faster

File: test/feature/AsyncGenerators/Demo.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    yield i;
+    ++i;
+  }
+}
+
+(async function() {
+  var count = 0;
+  for (value on asyncStream()) {
+    count += value;
+    if (value === 10) {
```
Comment (arv) [inline test/feature/AsyncGenerators/Demo.js]: reduce this too... we do not need more than 3 iterations

**Comment by arv** [comment]
Neat!

**Comment by mnieper** [comment]
Addressed your issues.

**Pull request merged**